store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -710,6 +710,7 @@ public void truncateDirtyFiles(long offsetToTruncate) {
             this.reputMessageService = new ConcurrentReputMessageService();
         }
 
+        setConfirmOffset(offsetToTruncate);
```

Comment:
Before merging #6618, confirmOffset had two different values in AutoSwitchHAService and commitlog.

Reply:
Yes, after #6618 is merged, we should use the DefaultMessageStore. 
I will help you review the #6618